### PR TITLE
Upgrade `@wordpress/data`

### PR DESCRIPTION
**Pull request by aaemnnosttv**
## Summary

Addresses issue #1769 

## Relevant technical choices

* JS tests are expected to fail initially as updates are still required to fix them

## Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [ ] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

**Comment by aaemnnosttv** [comment]
E2E jobs are all failing in the same (2) places as well

<details>
<summary>Summary</summary>

```
Summary of all failing tests
FAIL specs/modules/analytics/write-scope-requests.test.js (30.236s)
  ● Analytics write scope requests › prompts for additional permissions during a new Analytics profile creation if the user has not granted the Analytics edit scope

    TimeoutError: waiting for selector ".mdc-dialog--open" failed: timeout 3000ms exceeded

      234 | 		// Click on confirm changes button and wait for permissions modal dialog.
      235 | 		await submitForm();
    > 236 | 		await page.waitForSelector( '.mdc-dialog--open', { timeout: 3000 } );
          | 		           ^
      237 | 
      238 | 		// Click on proceed button and wait for oauth request.
      239 | 		await Promise.all( [

      at new WaitTask (../../node_modules/puppeteer/lib/DOMWorld.js:549:28)
      at DOMWorld._waitForSelectorOrXPath (../../node_modules/puppeteer/lib/DOMWorld.js:478:22)
      at DOMWorld.waitForSelector (../../node_modules/puppeteer/lib/DOMWorld.js:432:17)
      at Frame.waitForSelector (../../node_modules/puppeteer/lib/FrameManager.js:627:47)
      at Frame.<anonymous> (../../node_modules/puppeteer/lib/helper.js:112:23)
      at Page.waitForSelector (../../node_modules/puppeteer/lib/Page.js:1095:29)
      at _callee6$ (specs/modules/analytics/write-scope-requests.test.js:236:14)
      at tryCatch (../../node_modules/regenerator-runtime/runtime.js:45:40)
      at Generator.invoke [as _invoke] (../../node_modules/regenerator-runtime/runtime.js:274:22)
      at Generator.prototype.<computed> [as next] (../../node_modules/regenerator-runtime/runtime.js:97:21)
        -- ASYNC --
      at Frame.<anonymous> (../../node_modules/puppeteer/lib/helper.js:111:15)
      at Page.waitForSelector (../../node_modules/puppeteer/lib/Page.js:1095:29)
      at _callee6$ (specs/modules/analytics/write-scope-requests.test.js:236:14)
      at tryCatch (../../node_modules/regenerator-runtime/runtime.js:45:40)
      at Generator.invoke [as _invoke] (../../node_modules/regenerator-runtime/runtime.js:274:22)
      at Generator.prototype.<computed> [as next] (../../node_modules/regenerator-runtime/runtime.js:97:21)
      at asyncGeneratorStep (../../node_modules/@babel/runtime/helpers/asyncToGenerator.js:3:24)
      at _next (../../node_modules/@babel/runtime/helpers/asyncToGenerator.js:25:9)
          at runMicrotasks (<anonymous>)

FAIL specs/modules/search-console/dashboard-date-range.test.js (40.27s)
  ● date range filtering on dashboard views › loads new data when the date range is changed on a dashboard details view for a single post

    TimeoutError: Timeout exceeded while waiting for event

      120 | 			page.waitForNavigation(),
      121 | 			expect( postSearcher ).toClick( 'button', { text: /view data/i } ),
    > 122 | 			page.waitForResponse( ( res ) => res.url().match( 'google-site-kit/v1/data/' ) ),
          | 			     ^
      123 | 		] );
      124 | 
      125 | 		const TOTAL_IMPRESSIONS_28_DAYS = await getTotalImpressions();

      at Timeout.<anonymous> (../../node_modules/puppeteer/lib/helper.js:196:24)
        -- ASYNC --
      at Page.waitForResponse (../../node_modules/puppeteer/lib/helper.js:111:15)
      at _callee5$ (specs/modules/search-console/dashboard-date-range.test.js:122:9)
      at tryCatch (../../node_modules/regenerator-runtime/runtime.js:45:40)
      at Generator.invoke [as _invoke] (../../node_modules/regenerator-runtime/runtime.js:274:22)
      at Generator.prototype.<computed> [as next] (../../node_modules/regenerator-runtime/runtime.js:97:21)
      at asyncGeneratorStep (../../node_modules/@babel/runtime/helpers/asyncToGenerator.js:3:24)
      at _next (../../node_modules/@babel/runtime/helpers/asyncToGenerator.js:25:9)
          at runMicrotasks (<anonymous>)


Test Suites: 2 failed, 28 passed, 30 total
Tests:       2 failed, 94 passed, 96 total
```

</details>

File: assets/js/components/legacy-setup/wizard-progress-step.js [hunk 1]
```diff
@@ -20,7 +20,6 @@
  * External dependencies
  */
 import PropTypes from 'prop-types';
-import React from 'react';
```
Comment (tofumatt) [inline assets/js/components/legacy-setup/wizard-progress-step.js:23]: 😅

Comment (aaemnnosttv) [reply assets/js/components/legacy-setup/wizard-progress-step.js:23]: Yeah, oddly though I still saw hook errors after removing this until the change was added to the webpack config.

**Comment by techanvil** [comment]
Apologies for the force push (x2) - that was me fixing an error whereby I had pushed unsigned commits. I've now fixed my git config with `git config commit.gpgsign true` to ensure all commits are signed.

**Comment by techanvil** [comment]
I've made a bit of headway on this, but am putting it back down for now as I don't feel I can realistically tackle this with my current level of exposure to the codebase. It remains open for anyone to pick up.

**Comment by aaemnnosttv** [comment]
@tofumatt @techanvil would you please add your consent comments here to make @googlebot happy per above?

**Comment by techanvil** [comment]
@googlebot I consent.

**Comment by tofumatt** [comment]
@googlebot I consent.

**Comment by eugene-manuilov** [comment]
@googlebot I consent.

**Comment by techanvil** [comment]
Closing as this has been superseded by #6203

**Pull request closed without merging**